sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignDataView.java

```diff
@@ -50,21 +62,81 @@
 	private final CampaignDataGrid grid;
 	private CampaignFormDataFilterForm filterForm;
 
+	protected OptionGroup campaignFormElementImportance;
+
+	public static final String ONLY_IMPORTANT_FORM_ELEMENTS = "onlyImportantFormElements";
+
 	public CampaignDataView() {
 		super(VIEW_NAME);
 
 		criteria = ViewModelProviders.of(getClass()).get(CampaignFormDataCriteria.class);
 		grid = new CampaignDataGrid(criteria);
 
 		VerticalLayout mainLayout = new VerticalLayout();
-		mainLayout.addComponent(createFilterBar());
+		HorizontalLayout filtersLayout = new HorizontalLayout();
+
+		filtersLayout.setWidthFull();
+		filtersLayout.setMargin(false);
+		filtersLayout.setSpacing(true);
+
+		CampaignFormDataFilterForm filterBar = createFilterBar();
+		filtersLayout.addComponent(filterBar);
+		filtersLayout.setComponentAlignment(filterBar, Alignment.TOP_LEFT);
+		filtersLayout.setExpandRatio(filterBar, 0.8f);
+
+		createImportanceFilterSwitch();
+		filtersLayout.addComponent(campaignFormElementImportance);
+		filtersLayout.setComponentAlignment(campaignFormElementImportance, Alignment.TOP_RIGHT);
+		filtersLayout.setExpandRatio(campaignFormElementImportance, 0.2f);
+
+		mainLayout.addComponent(filtersLayout);
+
+		filterForm.getField(CampaignFormDataCriteria.CAMPAIGN_FORM_META).addValueChangeListener(e -> {
+			Object value = e.getProperty().getValue();
+			if (value == null) {
+				campaignFormElementImportance.setVisible(false);
+				campaignFormElementImportance.clear();
```

Comment:
Wouldn't it be better to keep the value and only set it to important as default, when the current value is empty?
When a user switches between forms he would probably like to keep the important/all setting as-is.